### DISCUSSION TITLE
How to set hls config options?

### DISCUSSION THREAD

**Discussion by bennlich** [muxinc/elements#576] [Q&A]
Hiya! I'm trying to figure out how to tweak some of the hls config options used by my mux-video player.

My first thought was that I might be able to pass them in as `<mux-video>` props, but I see [here](https://github.com/muxinc/elements/blob/main/packages/playback-core/src/index.ts#L185-L193) that only a select few get passed on to HLS.

My second thought was to use [Hls.DefaultConfig](https://github.com/video-dev/hls.js/blob/master/docs/API.md#hlsdefaultconfig-getset), but I'm loading mux-video with a script tag:

```
<script src="https://cdn.jsdelivr.net/npm/@mux/mux-video@0"></script>
```

and I can't figure out how to access the `Hls` object.

Is there a recommended way to do this?

**Comment by dylanjha** [comment]
Hi @bennlich 

The Hls.js internals on mux-player & mux-video is intentionally not part of the public API contract in order to maintain forward compatibility. We're keeping the option open for these packages in the future to do things like:

- Replace Hls.js with a different streaming engine
- Make a custom build of Hls.js that is slimmed down to only the pieces we need (which could potentially decrease the size of this package)

Having said that, the Hls.js instance is accessible via `document.querySelector('mux-player')._hls`.

If that helps you, then great! I just want to be clear that manipulating the Hls.js instance is not something we can make guarantees about in terms of forward compatibility.
